### PR TITLE
adding security API actions

### LINKED ISSUE #171: [FEATURE] Create opensearch security API
### Is your feature request related to a problem?
 OpenSearch client does not have security APIs

### What solution would you like?
  write an architecture similar to elasticsearch-ruby >=8 witcch have the API needed to send opensearch security requests (create user, create role...)

### PR DESCRIPTION
**Pull request by MarrakchiAyoub**
### Description
Added API actions for the security plugin

### Issues Resolved
Closes #171.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearch-api/lib/opensearch/api/actions/security/change_password.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+#
+# Licensed to OpenSearch B.V. under one or more contributor
```
Comment (dblock) [inline opensearch-api/lib/opensearch/api/actions/security/change_password.rb:10]: There's no such thing as OpenSearch B.V ;) This was an Elastic license grant which is not needed for all new code, remove.

File: opensearch-api/lib/opensearch/api/actions/security/change_password.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        #
+        # @option arguments [String] :username The username of the user to change the password for
+        # @option arguments [String] :refresh If `true` (the default) then refresh the affected shards to make this operation visible to search, if `wait_for` then wait for a refresh to make this operation visible to search, if `false` then do nothing with refreshes. (options: true, false, wait_for)
+        # @option arguments [Hash] :headers Custom HTTP headers
+        # @option arguments [String] :password the new password for the user (*Required*)
+        #
+        #
+        def change_password(arguments = {})
+          patch_user(arguments.merge({
+            username: arguments.delete(:username),
```
Comment (dblock) [inline opensearch-api/lib/opensearch/api/actions/security/change_password.rb:41]: Use `.fetch`. However here we merge `username` from `arguments` back to `arguments` which is a noop.

Can you please check whether we do `delete` all over the place? We should not be modifying hash data coming in. Open an issue if we do and don't have one open. See https://code.dblock.org/2020/11/11/dup-dont-modify-input-hash-parameters-in-ruby.html for an example of how this is a bad idea.

**Review by dblock — CHANGES_REQUESTED** [review]
Let's start with these.

**Comment by nhtruong** [comment]
Thanks for doing this @MarrakchiAyoub. Would you mind adding the [API specs](https://github.com/opensearch-project/opensearch-ruby/tree/main/opensearch-api/spec/opensearch/api/actions) for these actions as well?

**Comment by nhtruong** [comment]
This can now be closed since https://github.com/opensearch-project/opensearch-ruby/pull/179 has just been merged.

**Pull request closed without merging**